### ISSUE TITLE
Utilities.IO.NetworkX.load ndarray issue

### ISSUE THREAD

**Issue by joosthvanderlinden** [PMEAL/OpenPNM#674]
The nodes loop in `Utilities.IO.NetworkX.load` repeatedly assigns (and overwrites?) an `ndarray` to the `net` dictionary. I am seeing some undefined behaviour. If I run the code below once, only every second node is assigned its `props` values (zero elsewhere). If I run the code twice in a row, all nodes are filled.

I suspect this has something to do with `ndarray`'s buffer, but I am unable to find a solution.

To reproduce the problem:

```python
import yaml as _yaml
import scipy as _sp

with open('path/to/yaml/file.yaml', mode = 'r') as f:
    line = f.readline()
    a = _yaml.safe_load(f)

net = {}
Np = len(a['node'])
net.update({'pore.all': _sp.ones((Np,), dtype=bool)})
for n in a['node'].keys():
    props = a['node'][n]
    for item in props.keys():
        # Remove prepended pore. and pore_ if present
        for b in ['pore.', 'pore_']:
            item = item.replace(b, '')
        val = a['node'][n][item]
        dtype = type(val)
        if dtype is list:
            dtype = type(val[0])
            cols = len(val)
            net['pore.'+item] = _sp.ndarray((Np, cols), dtype=dtype, buffer = net['pore.'+item])
        else:
            net['pore.'+item] = _sp.ndarray((Np,), dtype=dtype)
        net['pore.'+item][n] = val
```

**Comment by jgostick** [comment]
For some reason, I did not get the message from Github that this issue was opened so forgive my delay.  I will take a look at it right away!

**Comment by jgostick** [comment]
Just to be clear, are you seeing this issue when you use the NetworkX IO class, or only when you do your own code? 

When I do the following, things work (using the yaml file in the test/fixtures dir):

```
import os
path = r"/path/to/file"
file = r"filename.yaml"
net = OpenPNM.Utilities.IO.NetworkX.load(os.path.join(path, file))
```

**Comment by jgostick** [comment]
First of all, I am totally baffled as to why the code in our NetworkX class works!  When I look at this it seems to me that it should over write the array in net['pore.property'] on each loop, but it doesn't.

To fix this I will adjust it to be:

```
net = {}
Np = len(a['node'])
net.update({'pore.all': _sp.ones((Np,), dtype=bool)})
for n in a['node'].keys():
    props = a['node'][n]
    for item in props.keys():
        # Remove prepended pore. and pore_ if present
        for b in ['pore.', 'pore_']:
            item = item.replace(b, '')

        val = a['node'][n][item]
        dtype = type(val)
        if item not in net.keys():
            if dtype is list:
                dtype = type(val[0])
                cols = len(val)
                net['pore.'+item] = _sp.ndarray((Np, cols), dtype=dtype)
            else:
                net['pore.'+item] = _sp.ndarray((Np,), dtype=dtype)
        net['pore.'+item][n] = val
```
Thirdly, the problem with your version above is that your calling net['pore.coords'] in the process of writing ['pore.coords'], hence the key not found error.  

Anyway, as I said, I'm still mystified how our version works, but I'll push my fix asap. 

Does this help?

**Comment by unimelbgeothermal** [comment]
Thanks @jgostick for looking into this. The `buffer = net['pore.'+item]` snippet in my original example was indeed incorrect (it snuck in after some experimenting). 

I tested your new fix on my `yaml` files and it works, so I'm happy with the fix.

**Comment by joosthvanderlinden** [comment]
Thanks @jgostick for looking into this. The `buffer = net['pore.'+item]` snippet in my original example was indeed incorrect (it snuck in after some experimenting). 

I tested your new fix on my `yaml` files and it works, so I'm happy with the fix.

**Comment by jgostick** [comment]
OK, great...we'll be merging the fix in soon.

**Comment by jgostick** [comment]
This was fixed in PR #675
